ui/src/app/pages/artifactVersion/components/modals/generateClientModal.tsx

```diff
@@ -0,0 +1,293 @@
+    GridItem,
+    Label,
+    Modal,
+    TextInput
+} from "@patternfly/react-core";
+import { CaretDownIcon } from "@patternfly/react-icons";
+import { ClientGeneration } from "../../../../../services";
+
+
+/**
+ * Properties
+ */
+export interface GenerateClientModalProps extends PureComponentProps {
+    artifactContent: string;
+    isOpen: boolean;
+    onClose: () => void;
+}
+
+/**
+ * State
+ */
+export interface GenerateClientModalState extends PureComponentState {
+    data: ClientGeneration;
+    languageIsExpanded: boolean;
+    isValid: boolean;
+    isErrorVisible: boolean;
+    downloadLink: string;
+    isDownloadLinkVisible: boolean;
+}
+
+export class GenerateClientModal extends PureComponent<GenerateClientModalProps, GenerateClientModalState> {
+
+    constructor(props: Readonly<GenerateClientModalProps>) {
+        super(props);
+    }
+
+    private downloadLink(): React.ReactElement {
+        console.log(this.state);
+        if (this.state.isDownloadLinkVisible) {
+            return <a
+                        key="DownloadLink"
+                        href={this.state.downloadLink}
+                        download={"client-" + this.state.data.language.toLowerCase() + ".zip"}
+                        onClick={this.onDownloadLinkClick}
+                    >
+                {this.state.isErrorVisible ? "Error" : "Download"}
+            </a>
+        } else if (this.state.isErrorVisible) {
+            return <Label key="Error">ERROR, please check the console logs for details.</Label>
+        } else {
+            return <div key="Placeholder"/>
+        }
+    }
+
+    public render(): React.ReactElement {
+        return (
+            <Modal
+                title="[EXPERIMENTAL] Client generation"
+                variant="large"
+                isOpen={this.props.isOpen}
+                onClose={this.props.onClose}
+                className="generate-client pf-m-redhat-font"
+                actions={[
+                    <Button key="Generate" variant="primary" data-testid="modal-btn-edit" onClick={this.doGenerate} isDisabled={!this.isGenerateEnabled()}>Generate</Button>,
+                    this.downloadLink()
+                ]}
+            >
+                <Form>
+                    <Grid hasGutter md={6}>
+                        <GridItem span={12}>
+                            <Label>[EXPERIMENTAL] This is "in-browser" generation of the client code using&nbsp;<a href="https://github.com/microsoft/kiota">Kiota</a>, please refer to&nbsp;
+                            <a href="https://microsoft.github.io/kiota/get-started/">the official documentation</a>&nbsp;to get started.</Label>
+                        </GridItem>
+
+                        <GridItem span={12}>
+                            <FormGroup
+                                label="Client Class Name"
+                                fieldId="form-client-name"
+                            >
+                                <TextInput
```

Comment:
I do think you should feature the include and exclude filters, or have some kind of endpoint selection mechanism. A big value proposition of Kiota is being able to generate specifically for the API surface the client cares about. Another aspect to that is that often some descriptions won't be the cleanest, and filtering out things that are failing is a quick way to get around and get things done.

Reply:
done 👍